architecture/2024-08-bicepdeployment-controller.md

```diff
@@ -0,0 +1,693 @@
+      ports: {
+        web: {
+          containerPort: port
+        }
+      }
+    }
+  }
+}
+```
+
+#### `app.bicepparam`
+```bicep
+using 'app.bicep'
+
+param application = ''
+param tag = ''
+param port = 3000
+```
+
+**Sample Output:**
+<!--
+Provide a sample output for the inputs provided above.
+-->
+```
+> rad bicep generate-kubernetes app.bicep --parameters @app.bicepparam --parameters tag=latest --parameters appnetworking.bicepparam --outfile app.yaml
+
+Generating BicepDeployment resource...
+BicepDeployment resource generated at app.yaml
+
+To apply the BicepDeployment resource onto your cluster, run:
+kubectl apply -f app.yaml
+```
+
+#### `app.yaml`
+```yaml
+kind: BicepDeployment
+apiVersion: radapp.io/v1alpha3
+metadata:
+  name: env
+  namespace: radius-system
+spec:
+  template: |
+    {
+      "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
+      "languageVersion": "2.1-experimental",
+      "contentVersion": "1.0.0.0",
+      "metadata": {
+        "_EXPERIMENTAL_WARNING": "This template uses ARM features that are experimental. Experimental features should be enabled for testing purposes only, as there are no guarantees about the quality or stability of these features. Do not enable these settings for any production usage, or your production environment may be subject to breaking.",
+        "_EXPERIMENTAL_FEATURES_ENABLED": [
+          "Extensibility"
+        ],
+        "_generator": {
+          "name": "bicep",
+          "version": "0.29.47.4906",
+          "templateHash": "14156924711842038952"
+        }
+      },
+      "parameters": {
+        "application": {
+          "type": "string"
+        },
+        "tag": {
+          "type": "string"
+        },
+        "port": {
+          "type": "int"
+        }
+      },
+      "imports": {
+        "Radius": {
+          "provider": "Radius",
+          "version": "latest"
+        }
+      },
+      "resources": {
+        "container": {
+          "import": "Radius",
+          "type": "Applications.Core/containers@2023-10-01-preview",
+          "properties": {
+            "name": "container",
+            "properties": {
+              "application": "[parameters('application')]",
+              "container": {
+                "image": "[format('ghcr.io/radius-project/samples/demo:{0}', parameters('tag'))]",
+                "ports": {
+                  "web": {
+                    "containerPort": "[parameters('port')]"
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  parameters: |
+    {
+      "application": {
+        "value": ""
+      },
+      "tag": {
+        "value": "latest"
+      },
+      "port": {
+        "value": 3000
+      }
+    }
+  bicepconfig: |
+    {
+      "imports": {
+        "Radius": {
+          "provider": "Radius",
+          "version": "latest"
+        }
+      }
+    }
+```
+
+### Bicep Module
```

Comment:
Can you describe why this example fits the user scenario? What is the user achieving through this bicep template?